app/models/integrations/base/publisher.rb

```diff
@@ -0,0 +1,29 @@
+module Integrations
+  module Base
+    class Publisher
+      include EpisodeSetOperations
+
+      attr_reader :show
+
+      def initialize(show:)
+        @show = show
+      end
+
+      def episodes_to_sync
+        filter_episodes_to_sync(show.episodes)
+      end
+
+      def episodes_to_archive
+        filter_episodes_to_archive(show.podcast_episodes, Set.new(show.episodes))
+      end
+
+      def episodes_to_unarchive
+        filter_episodes_to_unarchive(show.episodes)
+      end
+
+      def publish!
```

Comment:
This makes sense as it is the only part of this that I see called by another class/module, this seems like the interface of the publisher